frontend/pages/schemas/migrations/[schemaMigration].tsx

```diff
@@ -0,0 +1,147 @@
+import { ArrowBack } from '@mui/icons-material'
+import { Button, Container, Paper } from '@mui/material'
+import { useGetSchema } from 'actions/schema'
+import { putSchemaMigration, useGetSchemaMigrations } from 'actions/schemaMigration'
+import { useRouter } from 'next/router'
+import { useEffect, useState } from 'react'
+import Loading from 'src/common/Loading'
+import ErrorWrapper from 'src/errors/ErrorWrapper'
+import Link from 'src/Link'
+import SchemaMigrator from 'src/schemas/SchemaMigrator'
+import { CombinedSchema, QuestionMigration } from 'types/types'
+import { getErrorMessage } from 'utils/fetcher'
+import { getStepsFromSchema } from 'utils/formUtils'
+export default function SchemaMigrationEditor() {
+  const router = useRouter()
+  const { schemaMigration: schemaMigrationId }: { schemaMigration?: string } = router.query
+  const { schemaMigrations, isSchemaMigrationsLoading, isSchemaMigrationsError } =
+    useGetSchemaMigrations(schemaMigrationId)
+
+  const {
+    schema: sourceSchema,
+    isSchemaLoading: isSourceSchemaLoading,
+    isSchemaError: isSourceSchemaError,
+  } = useGetSchema(schemaMigrations[0] ? schemaMigrations[0].sourceSchema : '')
+  const {
+    schema: targetSchema,
+    isSchemaLoading: isTargetSchemaLoading,
+    isSchemaError: isTargetSchemaError,
+  } = useGetSchema(schemaMigrations[0] ? schemaMigrations[0].targetSchema : '')
```

Comment:
With a new end point (and new action `useGetSchemaMiagration`), we can move away from getting checking if we have the first item of an array.